### ISSUE TITLE
How to understand the commitment loss?

### ISSUE THREAD

**Issue by no-Seaweed** [EricGuo5513/momask-codes#67]
Hi Eric, 

Thank you for your great work. Now, I am attempting to understanding the following code 

`commit_loss = F.mse_loss(x, x_d.detach()) # It's right. the t2m-gpt paper is wrong on embed loss and commitment loss.`
`# Passthrough`
`x_d = x + (x_d - x).detach()`

It makes sense to calculate distance between gt x and feature after dequantizer x_d, but what is the purpose of recalculate x_d from (x_d -x).detach()? Why there is a detach here?

**Comment by Murrol** [comment]
Hi,
It helps pass (i.e. copy) the gradients from x_d to x.

**Comment by no-Seaweed** [comment]
Thank you for your quick response. Got it, thx. :)
